aws-distro-opentelemetry-node-autoinstrumentation/src/sampler/statistics.ts

```diff
@@ -8,7 +8,7 @@ export class Statistics implements ISamplingStatistics {
   public SampleCount: number;
   public BorrowCount: number;
 
-  constructor(requestCount: number = 0, sampleCount: number = 0, borrowCount: number = 0) {
```

Comment:
I feel like it make more sense to specify types here. We want people to know what parameters are expected without guessing. The same goes with function signatures. We want to rely on auto type inferring as much as possible but for those constructors, methods, functions, we want the type info as kind of documentation. 

Can we change the lint rule to enforce this? Maybe like this one: 

```
"@typescript-eslint/typedef": ["error", {
  "variableDeclaration": false,
  "parameter": true,
  "propertyDeclaration": true,
  "memberVariableDeclaration": true,
  "variableDeclarationIgnoreFunction": true
}],
```

Reference: https://typescript-eslint.io/rules/typedef/

Reply:
I'll probably simplify the PR. I'll switch back to previous logic, but enforce errors on `typedef` lint config. I'll just need to update some missing types for constants.